src/main/java/com/ibm/g11n/pipeline/client/ServiceClient.java

```diff
@@ -441,6 +442,111 @@ public abstract void updateResourceEntry(String bundleId, String language,
             String resKey, ResourceEntryDataChangeSet changeSet)
                     throws ServiceException;
 
+    //
+    // {serviceInstanceId}/v2/documents APIs
+    //
+
+    /**
+     * Returns a set of document IDs available in the service instance.
+     * <p>
+     * This operation is only allowed to {@link UserType#ADMINISTRATOR ADMINISTRATOR}
+     * of the service instance.
+     * 
+     * @param type The type of documents being requested.
+     * @return A set of document IDs.
+     * @throws ServiceException when the operation failed.
+     */
+    public abstract Set<String> getDocumentIds(DocumentType type) throws ServiceException;
+
+    /**
+     * Creates a new translatble document.
+     * <p>
+     * This operation is only allowed to {@link UserType#ADMINISTRATOR ADMINISTRATOR}
+     * of the service instance.
+     * 
+     * @param documentId
+     *          The new document's ID. Mut not be null or empty.
+     *          <br>The document ID must match a regular expression pattern
+     *          [a-zA-Z0-9][a-zA-Z0-9_.-]* and the length must be less than or equal
+     *          to 255.
+     * @param type The type of document being created.
+     * @param newDocumentData
+     *          The new document's configuration.
+     * @throws ServiceException when the operation failed.
+     */
+    public abstract void createDocument(String documentId, DocumentType type,
+            NewDocumentData newDocumentData) throws ServiceException;
+
+    /**
+     * Returns the document's configuration.
+     * <p>
+     * This operation is only allowed to all user types, but only
+     * basic information (source language and target languages) is included
+     * in the result if the requesting user type is {@link UserType#READER READER}.
+     * of the service instance.
+     * 
+     * @param documentId  The document ID. Must not be null or empty.
+     * @param type  The type of document being requested.
+     * @return          The document's configuration.
+     * @throws ServiceException when the operation failed.
+     */
+    public abstract DocumentData getDocumentInfo(String documentId, DocumentType type) throws ServiceException;
+    
+    /**
+     * Updates the document's configuration.
+     * <p>
+     * This operation is only allowed to {@link UserType#ADMINISTRATOR ADMINISTRATOR}
+     * of the service instance.
+     * 
+     * @param documentId  The document ID.
+     * @param type The type of document to be updated.
+     * @param changeSet The change set of document configuration.
+     * @throws ServiceException when the operation failed.
+     */
+    public abstract void updateDocument(String documentId, DocumentType type,
+            DocumentDataChangeSet changeSet) throws ServiceException;
+
+    /**
+     * Update document content in the source language.
```

Comment:
Hmm... I prefer to state - update document content in the specified language - and explain the language other than source language is not supported at this moment. And XXXException is thrown when a language other than a source language is specified.